src/main/java/com/google/idea/perf/TracerController.kt

```diff
@@ -174,6 +174,17 @@ class TracerController(
             TracerConfig.traceMethods(classJvmName, methodName)
             retransformClasses(setOf(className))
         }
+        else if (cmd.startsWith("untrace")) {
+            val spec = cmd.substringAfter("untrace").trim()
+            val split = spec.split('#')
+            if (split.size != 2) {
+                LOG.warn("Invalid untrace format; expected com.example.Class#method")
+                return
+            }
+            val (className, methodName) = split
+            val classJvmName = className.replace('.', '/')
+            TracerConfig.untraceMethods(classJvmName, methodName)
```

Comment:
I imagine the main use case for `untrace` is when the user has accidentally traced a very high-traffic method like `String.charAt()`, and wants to undo the tracing in order to remove the overhead.

So, it could be nice to call `retransformClasses(setOf(className))` after this line in order to restore the method to its original bytecode and thus remove tracing overhead entirely.

(Note, I guess we would also need to edit `TraceConfig.getMethodId()` to return `null` when the tracepoint is 'disabled', so that `TracerMethodTransformer` does not re-insert the tracing hooks when the class is retransformed.)

What do you think? I'm OK with either approach.

Reply:
Yeah I agree, we should undo bytecode modifications since this a common use case of hte `untrace` command.